cluster_test.go

```diff
@@ -66,20 +72,28 @@ func newTestCluster(t *testing.T) *testCluster {
 	root, err := ioutil.TempDir("", "sequins-cluster-")
 	require.NoError(t, err)
 
-	zkServers, _ := createTestZkCluster(t)
+	zkServers, zkCluster := createTestZkCluster(t)
+
+	// We have a specific transport to the client, so it doesn't try to reuse
+	// connections between tests
+	var testClient = &http.Client{
+		Timeout:   100 * time.Millisecond,
+		Transport: &http.Transport{},
+	}
 
 	return &testCluster{
-		T:         t,
-		binary:    binary,
-		root:      root,
-		sequinses: make([]*testSequins, 0),
-		zkServers: zkServers,
+		T:          t,
+		binary:     binary,
+		root:       root,
+		sequinses:  make([]*testSequins, 0),
+		zkServers:  zkServers,
+		zkCluster:  zkCluster,
+		testClient: testClient,
 	}
 }
 
 func (tc *testCluster) addSequins() {
-	index := len(tc.sequinses)
-	port := 19590 + index // TODO pick random high port
+	port := randomPort()
```

Comment:
Fair, I'd mostly be worried about interfering with anything else that's running, but if the tests fail without interrupting other processes, I'm good with it.